iothub/service/src/MessageSystemPropertyNames.cs

```diff
@@ -36,6 +36,9 @@ public static class MessageSystemPropertyNames
         /// </summary>
         public const string To = "to";
 
+        /// <summary>
+        /// The maximum number of times a message can transition between the Enqueued and Invisible states.
+        /// After the maximum number of transitions, the IoT hub sets the state of the message to dead lettered.
         /// </summary>
         public const string DeliveryCount = "iothub-deliverycount";
```

Comment:
does DeliveryCount hold the max allowed value, or the current count of the no of times the message has transitioned between enqueued and invisible?
It might also help to add a link to the ms doc, so that users can visualize the entire lifecycle: https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-c2d#the-cloud-to-device-message-life-cycle

Reply:
I think it should not be the maximum thinking about it. I should just be the count and the logic to compare against the max should be in the hub.